build.rs

```diff
@@ -199,96 +203,86 @@ const RING_TEST_HEADERS: &'static [&'static str] =
       "crypto/test/file_test.h",
       "crypto/test/bn_test_util.h"];
 
-const RING_PERL_INCLUDES: &'static [&'static str] = &["crypto/sha/asm/sha-x86_64.pl"];
+const RING_PERL: &'static [&'static str] = &["crypto/sha/asm/sha-x86_64.pl"];
 
 const RING_MAKEFILES: &'static [&'static str] = &["Makefile",
                                                   "mk/top_of_makefile.mk",
                                                   "mk/bottom_of_makefile.mk",
                                                   "mk/ring.mk"];
 
+const C_FLAGS: &'static [&'static str] = &["-std=c1x",
+                                           "-Wbad-function-cast",
+                                           "-Wmissing-field-initializers",
+                                           "-Wmissing-prototypes",
+                                           "-Wnested-externs",
+                                           "-Wstrict-prototypes"];
+
+const CPP_FLAGS: &'static [&'static str] = &["-std=c++0x",
```

Comment:
I noticed you copied `CFLAGS` and `CPPFLAGS` from mk/top_of_makefile.mk, but you didn't copy `CXXFLAGS`. Instead of looks like you confused `CPPFLAGS` and `CXXFLAGS`. `CPPFLAGS` should be passed to every compilation (*.S, *.c, *.cc), `CFLAGS` should additionally be passed for *.c only, and `CXXFLAGS` should be passed for *.cc.

In particular, `-std=c++0x` should be in `CXXFLAGS` but most of the other stuff here should remain in `CPPFLAGS`.